components/utils/src/vec.rs

```diff
@@ -0,0 +1,44 @@
+pub trait InsertMany {
+    type Element;
+    fn insert_many(&mut self, elem_to_insert: Vec<(usize, Self::Element)>);
+}
+
+impl<T> InsertMany for Vec<T> {
+    type Element = T;
+
+    /// Efficiently insert multiple element in their specified index.
+    /// The index should be sorted in ascending order.
```

Comment:
you mean something like "the elements should be sorted in ascending order by their index"?

Reply:
Fixed